stl/inc/format

```diff
@@ -302,69 +418,76 @@ constexpr const _CharT* _Parse_format_specs(const _CharT* _Begin, const _CharT*
-    basic_format_parse_context(const basic_format_parse_context&) = delete;
-    basic_format_parse_context& operator=(const basic_format_parse_context&) = delete;
-
-    _NODISCARD constexpr const_iterator begin() const noexcept {
-        return _Format_string.begin();
-    }
-    _NODISCARD constexpr const_iterator end() const noexcept {
-        return _Format_string.end();
-    }
-    constexpr void advance_to(const const_iterator _It) {
-        _Adl_verify_range(_It, _Format_string.end());
-        const auto _Diff = static_cast<size_t>(_It._Unwrapped() - _Format_string._Unchecked_begin());
-        _Format_string.remove_prefix(_Diff);
+template <class _CharT, _Parse_replacement_field_callbacks<_CharT> _HandlerT>
+constexpr const _CharT* _Parse_replacement_field(const _CharT* _Begin, const _CharT* _End, _HandlerT&& _Handler) {
+    ++_Begin;
+    if (_Begin == _End) {
+        throw format_error("Invalid format string.");
     }
 
-    // While the standard presents an exposition only enum value for
-    // the indexing mode (manual, automatic, or unknown) we use _Next_arg_id to indicate it.
-    // _Next_arg_id == 0 means unknown
-    // _Next_arg_id > 0 means automatic
-    // _Next_arg_id == -1 means manual
-    constexpr size_t next_arg_id() {
-        if (_Next_arg_id == -1) {
-            throw format_error("Can not switch from manual to automatic indexing");
+    if (*_Begin == '}') {
+        // string was "{}", and we have a replacement field
+        _Handler._On_replacement_field(_Handler._Parse_context.next_arg_id(), _Begin);
+    } else if (*_Begin == '{') {
+        // string was "{{", so we have a literal "{" to print
+        _Handler._On_text(_Begin, _Begin + 1);
+    } else {
+        _Id_adapter<_CharT> _Adapter{_Handler._Parse_context};
+        _Begin     = _Parse_arg_id(_Begin, _End, _Adapter);
+        _CharT _Ch = _CharT{};
+        if (_Begin != _End) {
+            _Ch = *_Begin;
+        }
+        if (_Ch == '}') {
+            _Handler._On_replacement_field(_Adapter._Arg_id, _Begin);
+        } else if (_Ch == ':') {
+            _Begin = _Handler._On_format_specs(_Adapter._Arg_id, _Begin + 1, _End);
+            if (_Begin == _End || *_Begin != '}') {
+                throw format_error("Unknown format specifier.");
+            }
+        } else {
+            throw format_error("Missing '}' in format string.");
         }
-
-        return static_cast<size_t>(_Next_arg_id++);
     }
+    return _Begin + 1;
+}
 
-    constexpr void check_arg_id(const size_t _Id) {
-        if (_STD is_constant_evaluated()) {
-            if (_Id >= _Num_args) {
-                _You_see_this_error_because_arg_id_is_out_of_range();
+template <class _CharT, _Parse_replacement_field_callbacks<_CharT> _HandlerT>
+constexpr void _Parse_format_string(basic_string_view<_CharT> _Format_str, _HandlerT&& _Handler) {
+    auto _Begin = _Format_str.data();
+    auto _End   = _Begin + _Format_str.size();

```

Comment:
I think it would be clearer if we would do
```suggestion
    auto _Begin = _Format_str._Unchecked_begin();
    auto _End   = _Format_str._Unchecked_end();;
```

Reply:
it's minor so I'll keep as is, just to preserve similarity with libfmt (and because I wanna get this pr merged)